pkg/printers/template_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+func TestTemplate(t *testing.T) {
+	testCase := map[string]struct {
+		template  string
+		obj       runtime.Object
+		expectOut string
+		expectErr string
+	}{
+		"support base64 decoding of secret data": {
+			template: "{{ .Data.username | b64dec }}",
+			obj: &api.Secret{
+				Data: map[string][]byte{
+					"username": []byte("hunter"),
+				},
+			},
+			expectOut: "hunter",
+		},
+		"test error path for base64 decoding": {
+			template:  "{{ .Data.username | b64dec }}",
+			obj:       &badlyMarshaledSecret{},
+			expectErr: "error executing template \"{{ .Data.username | b64dec }}\": template: output:1:20: executing \"output\" at <b64dec>: error calling b64dec: illegal base64 data at input byte 0",
```

Comment:
This relies on explicit error strings from several packages in the Go std library, which may change depending on the Go version used to run the test. This needs to only rely on error strings we control (e.g. check for a substring in the error), or only test for the presence of an error.

Reply:
OK. I'll see what I can come up with. Thanks!